File: zzWikiUnitOfWorkVB/Backup/Router.ascx.vb
'
' DotNetNuke® - http://www.dotnetnuke.com
' Copyright (c) 2002-2012
' by DotNetNuke Corporation
'
' Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
' documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
' the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
' to permit persons to whom the Software is furnished to do so, subject to the following conditions:
'
' The above copyright notice and this permission notice shall be included in all copies or substantial portions 
' of the Software.
'
' THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
' TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
' THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
' CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
' DEALINGS IN THE SOFTWARE.
'

Imports DotNetNuke.Entities.Modules
Imports DotNetNuke.Services.Localization
Imports DotNetNuke.Entities.Modules.Actions
Imports DotNetNuke.Modules.Wiki.Entities

Namespace DotNetNuke.Modules.Wiki

    Partial Class Router
        Inherits WikiControlBase
        'Implements ISearchable
        'Implements IPortable
        Implements IActionable


#Region " Web Form Designer Generated Code "

        'This call is required by the Web Form Designer.
        <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()

        End Sub

        Private Sub Page_Init(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Init
            'CODEGEN: This method call is required by the Web Form Designer
            'Do not modify it using the code editor.
            InitializeComponent()
        End Sub

#End Region

        Private Sub Router_Page_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

            'load the menu on the left
            Dim leftControl As String = "Controls/WikiMenu.ascx"
            Dim mbl As WikiControlBase = CType(TemplateControl.LoadControl(leftControl), WikiControlBase)
            mbl.ModuleConfiguration = ModuleConfiguration
            mbl.ID = System.IO.Path.GetFileNameWithoutExtension(leftControl)
            phWikiMenu.Controls.Add(mbl)

            Dim controlToLoad As String = GetControlString(Request.QueryString("loc"))
            Dim wikiContent As WikiControlBase = CType(LoadControl(controlToLoad), WikiControlBase)
            wikiContent.ModuleConfiguration = ModuleConfiguration
            wikiContent.ID = System.IO.Path.GetFileNameWithoutExtension(controlToLoad)
            phWikiContent.Controls.Add(wikiContent)


            If (controlToLoad.ToLower <> "edit.ascx") Then
                Dim buttonControlToLoad As String = "Controls/WikiButton.ascx"
                Dim wikiButton As WikiControlBase = CType(LoadControl(buttonControlToLoad), WikiControlBase)

                wikiButton.ModuleConfiguration = ModuleConfiguration
                wikiButton.ID = System.IO.Path.GetFileNameWithoutExtension(buttonControlToLoad)
                phWikiContent.Controls.Add(wikiButton)
            End If

            'print
            Dim objAction As ModuleAction

            For Each objAction In Actions
                If (objAction.CommandName = ModuleActionType.PrintModule) Then
                    objAction.Url += "&topic=" + WikiData.EncodeTitle(PageTopic)
                End If
            Next

        End Sub

        Private Function GetControlString(ByVal loc As String) As String
            If loc Is Nothing Then
                Return "Start.ascx"
            Else
                Select Case (loc.ToLower())
                    Case "start"
                        Return "Start.ascx"
                    Case "edit"
                        Return "Edit.ascx"
                    Case "topichistory"
                        Return "topichistory.ascx"
                    Case "search"
                        Return "search.ascx"
                    Case "recentchanges"
                        Return "recentchanges.ascx"
                    Case "index"
                        Return "controls/index.ascx"
                    Case Else
                        Return "start.ascx"
                End Select
            End If
        End Function



        'Private Sub ImageButton1_Click(ByVal sender As System.Object, ByVal e As System.Web.UI.ImageClickEventArgs) Handles ImageButton1.Click
        '    If ShowNav Then

        '        Me.ImageButton1.AlternateText = Localization.GetString("ShowNavigation", LocalResourceFile) ' "Show Navigation"
        '        Me.ImageButton1.ImageUrl = TemplateSourceDirectory + "/images/ShowNav.gif"
        '        Me.LinksPanel.Visible = False
        '        ShowNav = False
        '        Me.Session("wiki" + ModuleId.ToString + "ShowNav") = False
        '    Else
        '        Me.ImageButton1.AlternateText = Localization.GetString("HideNavigation", LocalResourceFile) '"Hide Navigation"
        '        Me.ImageButton1.ImageUrl = TemplateSourceDirectory + "/images/HideNav.gif"
        '        Me.LinksPanel.Visible = True
        '        ShowNav = False
        '        Me.Session("wiki" + ModuleId.ToString + "ShowNav") = True
        '    End If
        'End Sub


        Public ReadOnly Property ModuleActions() As DotNetNuke.Entities.Modules.Actions.ModuleActionCollection Implements DotNetNuke.Entities.Modules.IActionable.ModuleActions
            Get
                Dim Actions As New DotNetNuke.Entities.Modules.Actions.ModuleActionCollection
                Actions.Add(GetNextActionID, Localization.GetString("Administration", LocalResourceFile).ToString(), "", "", "", EditUrl("Administration"), False, Security.SecurityAccessLevel.Admin, True, False)
                Return Actions
            End Get
        End Property

    End Class
End Namespace

File: Views/Router.ascx.cs
#region Copyright

//--------------------------------------------------------------------------------------------------------
// <copyright file="Router.ascx.cs" company="DNN Corp®">
//      DNN Corp® - http://www.dnnsoftware.com Copyright (c) 2002-2013 by DNN Corp®
//
//      Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
//      associated documentation files (the "Software"), to deal in the Software without restriction,
//      including without limitation the rights to use, copy, modify, merge, publish, distribute,
//      sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
//      furnished to do so, subject to the following conditions:
//
//      The above copyright notice and this permission notice shall be included in all copies or
//      substantial portions of the Software.
//
//      THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
//      NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
//      NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
//      DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
//      OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
// </copyright>
////--------------------------------------------------------------------------------------------------------

#endregion Copyright

using DotNetNuke.Entities.Modules;
using DotNetNuke.Entities.Modules.Actions;
using DotNetNuke.Services.Exceptions;
using DotNetNuke.Services.Localization;
using DotNetNuke.Wiki.Utilities;
using System;
using System.Web.UI;

namespace DotNetNuke.Wiki.Views
{
    /// <summary>
    /// Router Class based on the WikiModuleBase Class
    /// </summary>
    internal partial class Router : WikiModuleBase, IActionable
    {
        #region Constructor

        /// <summary>
        /// Initializes a new instance of the <see cref="Router"/> class.
        /// </summary>
        public Router()
        {
            this.Load += this.Router_Page_Load;
        }

        #endregion Constructor

        #region Properties

        /// <summary>
        /// Gets the module actions.
        /// </summary>
        /// <value>The module actions.</value>
        public DotNetNuke.Entities.Modules.Actions.ModuleActionCollection ModuleActions
        {
            get
            {
                DotNetNuke.Entities.Modules.Actions.ModuleActionCollection actions = new DotNetNuke.Entities.Modules.Actions.ModuleActionCollection();
                actions.Add(
                    this.GetNextActionID(),
                    Localization.GetString("Administration", LocalResourceFile).ToString(),
                    string.Empty,
                    string.Empty,
                    string.Empty,
                    this.EditUrl("Administration"),
                    false,
                    Security.SecurityAccessLevel.Admin,
                    true,
                    false);
                return actions;
            }
        }

        #endregion Properties

        #region Events

        /// <summary>
        /// Handles the Load event of the Router_Page control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="System.EventArgs"/> instance containing the event
        /// data.</param>
        private void Router_Page_Load(object sender, System.EventArgs e)
        {
            try
            {
                // Load the menu on the left
                string leftControl = "SharedControls//WikiMenu.ascx";
                WikiModuleBase wikiModuleBase = (WikiModuleBase)TemplateControl.LoadControl(leftControl);
                wikiModuleBase.ModuleConfiguration = this.ModuleConfiguration;
                wikiModuleBase.ID = System.IO.Path.GetFileNameWithoutExtension(leftControl);
                phWikiMenu.Controls.Add(wikiModuleBase);

                string controlToLoad = this.GetControlString(Request.QueryString["loc"]);
                WikiModuleBase wikiContent = (WikiModuleBase)LoadControl(controlToLoad);
                wikiContent.ModuleConfiguration = this.ModuleConfiguration;
                wikiContent.ID = System.IO.Path.GetFileNameWithoutExtension(controlToLoad);
                phWikiContent.Controls.Add(wikiContent);

                if (controlToLoad.ToLower().Equals("start.ascx"))
                {
                    string buttonControlToLoad = "SharedControls//WikiButton.ascx";
                    WikiModuleBase wikiButton = (WikiModuleBase)LoadControl(buttonControlToLoad);

                    wikiButton.ModuleConfiguration = this.ModuleConfiguration;
                    wikiButton.ID = System.IO.Path.GetFileNameWithoutExtension(buttonControlToLoad);
                    phWikiContent.Controls.Add(wikiButton);
                }

                // Print the Topic
                foreach (ModuleAction objAction in this.Actions)
                {
                    if (objAction.CommandName.Equals(ModuleActionType.PrintModule))
                    {
                        objAction.Url += "&topic=" + WikiMarkup.EncodeTitle(this.PageTopic);
                    }
                }
            }
            catch (Exception exc)
            {
                Exceptions.ProcessModuleLoadException(this, exc);
            }
        }

        #endregion Events

        #region Methods

        /// <summary>
        /// Gets the control string.
        /// </summary>
        /// <param name="locationString">The location.</param>
        /// <returns>Control Name</returns>
        private string GetControlString(string locationString)
        {
            if (locationString == null)
            {
                return "Start.ascx";
            }
            else
            {
                switch (locationString.ToLower())
                {
                    case "start":
                        return "Start.ascx";

                    case "edit":
                        return "Edit.ascx";

                    case "topichistory":
                        return "topichistory.ascx";

                    case "search":
                        return "search.ascx";

                    case "recentchanges":
                        return "recentchanges.ascx";

                    case "index":
                        return "SharedControls//index.ascx";

                    default:
                        return "start.ascx";
                }
            }
        }

        #endregion Methods
    }
}